e2etest/localserviceimpact_test.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2etest
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+)
+
+var _ = Describe("Local service not impacted", func() {
+	helloDeployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hello",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicaCount,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "hello",
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "hello"},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "hello-tcp",
+							Image: "alpine/socat:1.7.4.4",
+							Args:  []string{"-v", "-v", "TCP-LISTEN:42,crlf,reuseaddr,fork", "SYSTEM:echo $(CLUSTER_ID)"},
+							Env: []v1.EnvVar{
+								{
+									Name: "CLUSTER_ID",
+									ValueFrom: &v1.EnvVarSource{
+										ConfigMapKeyRef: &v1.ConfigMapKeySelector{
+											LocalObjectReference: v1.LocalObjectReference{
+												Name: "cluster-info",
+											},
+											Key: "clusterID",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	var (
+		namespace string
+
+		ctx    = context.Background()
+		reqPod *v1.Pod
+	)
+
+	BeforeEach(func() {
```

Comment:
This is fine as-is for now, but one of the design goals we (Paul Morie, @lauralorenz and I) had for the compliance tests was to come up with at least a library of functions to set up and tear down test scenarios. This would be a good start once extracted into a function (or several).

Ultimately we could have a DSL for test scenarios but I think it’s best to evolve that from what actually ends up being useful :wink:.

(Just noting for future reference, please don’t change anything as a result of this comment.)

Reply:
Related to your other comment, I did extract the "export a service from this cluster to that cluster" part of this into its own function.